### DISCUSSION TITLE
Thank you!

### DISCUSSION THREAD

**Discussion by prmadev** [zk-org/zk#177] [Share]
I just wanted to let you know that zk inspired me to learn go. 
And this is the first programming language that I actually spent a good time on. Before this I have been tried python and JS  but they were not for me. I never considered go, because I thought it was that frugal language that google used internally and only suits the backend of huge data centers. Boy! How wrong was I? 
When I discovered zk and played around with it for a few days, I enjoyed it so much so that I got curious about the code. And I took a look at it and got more and more curious. And now I'm in the middle of my second udemy course of go, doing some exercism and a lot of go time podcasts and I am excited to become fluent enough so that I can start familiarizing myself to the zk code base! 
Thank you so much for your work!

**Comment by mickael-menu** [comment]
`zk` was also my first experience with Go! I was looking for a language that made cross-platform easy, fast and without dependencies. Since then I've been using Go for all my CLI tools.

One of my favorite features of Go is the implicit interfaces, which is a like a safer duck typing: https://hackernoon.com/why-i-like-gos-interfaces-2891adf2803c And now that Go 1.18 is released with support for generics, I will eventually be able to simplify a lot of things in `zk`'s codebase.

**Comment by jvaverka** [comment]
While your choice of using Go for this project is interesting, I'd just like to say thank you and appreciate the end result. For some reason `zk` just clicks for me, and after a couple weeks I am experiencing many productivity benefits. Best of all, I know as I continue to learn & grow, `zk` will grow with me without slowing me down. So, thank you!

**Reply by mickael-menu** [reply]
Thank you for your kind words!

**Comment by dzintars** [comment]
For couple of years i was looking for a such tool. Lately i was using Obsidian. But switching from Nvim to Electron app is a pain. Even if that is only few keystrokes away. :)
Most critical feature for me was the ability to search files by aliases or titles. For example `aliases: [Neovim, Nvim]` would return me a same file for booth search terms. You have no this feature, but `title` also is great enough option!
I don't get why people care about file names or even worst - directory structure. IMO that leads to a huge pain in the long term when you will decide to restructure your zettel. Or to rename the note.
So.. whatever.
I think you really nailed the feature set and main concept of this tool. `zk-nvim` also is so helpful!
So... BIG, BIG Thank You for making this!!!

**Reply by mickael-menu** [reply]
Thank you fellow Colemakian! 🙂 

> Most critical feature for me was the ability to search files by aliases or titles. For example `aliases: [Neovim, Nvim]` would return me a same file for booth search terms. You have no this feature, but `title` also is great enough option!

`zk` supports `aliases` but only for the [`--mention` and `--mentioned-by` flags](
https://github.com/mickael-menu/zk/blob/4b76fbadf171dbb18a3154701fe3b4d1cba31a0c/docs/note-filtering.md#locate-mentions-of-other-notes). How exactly would you use it to search for a note? Is it in the `:ZkNotes` screen of `zk-nvim`? If yes I suggest you open a feature request there.

**Reply by dzintars** [reply]
Hi. 😉 
I don't think you should think about this. I believe `aliases` handling is just Obsidian specific thing.
For me `title` works well enough, and i don't use `aliases` THAT often. I can live without this feature for sure.
But, yes it's about `:ZkNotes` and LSP autocompletion when linking to note. I can look up for note either by `Neovim` or by `Nvim`.
It's just a sugar (nice to have).

**Reply by mickael-menu** [reply]
`:ZkNotes` would require some work in `zk-nvim`, but for LSP autocompletion it should already be possible.

You need this in your `zk` config file:

```toml
[lsp.completion]
note-filter-text = "{{title}} {{path}} {{metadata.aliases}}"
```

This is the string that will be used when filtering in the completion pop-up. Note that not all LSP completion clients support `filterText`, if it doesn't work for you, you can try:

```toml
[lsp.completion]
note-label = "{{title}} {{metadata.aliases}}"
```

The only difference is that the aliases will be visible in the completion list with `note-label`.

As you can see, you can put whatever you need to complete on in `note-filter-text`. Take a look at https://github.com/mickael-menu/zk/blob/main/docs/config-lsp.md

**Reply by jvaverka** [reply]
What you describe above for aliases + LSP autocompletion suddenly made aliases _much_ more relevant IMO. Very cool!

> Thank you fellow Colemakian!

I also use Colemak-DH and Neovim.. Small world.

**Reply by dzintars** [reply]
Yeah Colemak is really nice.
![image](https://user-images.githubusercontent.com/547420/171877194-c6185ad5-f0d6-44ac-8a54-a8fb31133db6.png)
But i still shaping my layout. Can't get my alternate hand motions right for special key binds.
Sorry for off-topic. 😏

**Reply by mickael-menu** [reply]
Same I suck at hitting the top row. 😄  I created a layout where all the important programming symbols are on the home row when I press ALT. I've been using this for the past 10 years and don't have any wrist pain anymore!

<img width="597" alt="Screenshot 2022-06-03 at 18 08 34" src="https://user-images.githubusercontent.com/58686775/171904127-01d5e42b-ed40-4604-b578-3639ecc33742.png">

**Comment by prmadev** [comment]
I once learned colmak. It was perfect. The fatal flaw was with
compatibility, or when i was using systems other than my own. Though it was
ten years ago. Is it better these days on the compatibility front?

**Reply by mickael-menu** [reply]
It's still a pain, thankfully I work from home and I don't have to use other machines than my own! On the plus side, people are lost when on my computer. 😄

**Reply by dzintars** [reply]
![Keebio-Iris-KAT-blank](https://user-images.githubusercontent.com/547420/171930057-423b068f-9938-40f3-b543-619adb9a024c.jpeg)
Definitely! 😄 Colemak+Blanks+Tiling window manager are safer than any lock screen. 😄
